api/include/opentelemetry/std/span.h

```diff
@@ -60,7 +60,7 @@ OPENTELEMETRY_END_NAMESPACE
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace nostd
 {
-constexpr std::size_t dynamic_extent = (std::numeric_limits<std::size_t>::max)();
+constexpr std::size_t dynamic_extent = std::numeric_limits<std::size_t>::max();
```

Comment:
Why was this changed back, this was just fixed for NOMINMAX in windows ?

Reply:
oh this was added after I approved the PR :)

Reply:
Apologizes for this. I was trying to see if the new CI could catch this kind of issue. Will revert this before merging.

Reply:
Converted this PR to draft for now.

Reply:
It is verified that the new CI pipeline can catch the issue. See the CI result below which reproduces the issue.

https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/7204221666/job/19625317139?pr=2450

Reply:
Thanks for validating this.